app/models/form.rb

```diff
@@ -110,6 +107,25 @@ def filter_conditional_questions(questions, case_log)
     applicable_questions
   end
 
+  def page_routed_to?(page, case_log)
```

Comment:
This could be simplified quite a lot if we changed the form definition schema so that the conditions for showing a page were attached to the question it affects rather than to the question providing the data. Given the size of this PR I think that should be a subsequent step though.

Reply:
Now done